WeakAuras/Profiling.lua

```diff
@@ -298,15 +338,72 @@ function WeakAuras.ToggleProfile()
   end
 end
 
-local function PrintOneProfile(name, map, total)
+local function autoStartStopProfiling(event)
+  if event == "ENCOUNTER_START" or event == "PLAYER_REGEN_DISABLED" then
+    WeakAuras.StartProfile()
+  elseif event == "ENCOUNTER_END" or event == "PLAYER_REGEN_ENABLED" then
+    WeakAuras.StopProfile()
+      if event == "ENCOUNTER_END" then
+        profilePopup:UnregisterEvent("ENCOUNTER_START")
+        profilePopup:UnregisterEvent("ENCOUNTER_END")
+      else
+        profilePopup:UnregisterEvent("PLAYER_REGEN_DISABLED")
+        profilePopup:UnregisterEvent("PLAYER_REGEN_ENABLED")
+      end
+  end
+end
+
+registerProfile = function(startType)
+  if startType == "boss" then startType = "encounter" end
+  if currentlyQueuedProfile and (startType == currentlyQueuedProfile) then
```

Comment:
I don't think WeakAuras.StartProfile() should ever cancel a profiling. That is the buttons should not call StartProfile if they want to cancel a queued profiling.